rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/domain/service/LRAService.java

```diff
@@ -363,10 +370,10 @@ public int joinLRA(StringBuilder recoveryUrl, URI lra, long timeLimit,
 
                         if (!LRAConstants.AFTER.equals(rel)) {
                             // participants are not allowed to join inactive LRAs
-                            return Response.Status.PRECONDITION_FAILED.getStatusCode();
+                            return Response.Status.GONE.getStatusCode(); // report it as GONE
```

Comment:
I am also curious to learn more about this. I assume that the lra is already completed and so inactive, in that case the GONE response is more appropriate according to https://github.com/eclipse/microprofile-lra/blob/main/spec/src/main/asciidoc/microprofile-lra-spec.asciidoc?plain=1#L302-L304

Reply:
I've just successfully reran the tests without the change so I'll revert this bit. But, for completeness, here is the rationale for the change:

I think it's what the code was returning before, it may or may not have been going through this catch block: https://github.com/jbosstm/narayana/pull/2331/files#diff-34b2877655e9cabf2c9bbe0e080b7c5060095f760099cee6dc74b6e4ff57cb65L783, also there is an eclipse microprofile spec test that tends to agree with this view (https://github.com/eclipse/microprofile-lra/blob/2.0/tck/src/main/java/org/eclipse/microprofile/lra/tck/TckTests.java#L361). And the analysis I did on JBTM-3331 (https://issues.redhat.com/browse/JBTM-3331?focusedId=25955017&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-25955017) back in 2020 indicated that during the spec process we switched from using 404 Not Found preferring instead to use 410 Gone or 412 Precondition Failed.

Reply:
I have just pushed that change. I also added an extra commit to fix a confusing formatting problem in NarayanaLRAClient#enlistCompensator